### ISSUE TITLE
Support more HTML5 input "types" for better end-user UX

### ISSUE THREAD

**Issue by FuelAgency** [rhythmagency/formulate#135]
HTML5 implements a range of new form input types that make filling out forms a much friendlier experience, especially for mobile device users. When a text field is selected, a customised keyboard is shown that is optimised to the field requirements.

- An "Email" field displays a keyboard with an @ symbol
- A Number field shows a number keypad
- A Colour field shows a colour picker

There's some great improvements available (device and browser dependant). It would be nice if the Formulate tool began to implement some of them.

See documentation here: https://www.w3schools.com/html/html_form_input_types.asp

**Comment by Nicholas-Westby** [comment]
Makes sense. I've marked this as "help wanted" to denote that anybody in the community can submit a pull request for this.

BTW, for the fields that are essentially just a text field with a different type (e.g., email), it might make sense to just modify the text field so that it has a type drop down.

**Comment by FuelAgency** [comment]
> BTW, for the fields that are essentially just a text field with a different type (e.g., email), it might make sense to just modify the text field so that it has a type drop down.

Indeed. The "Text" variants would seem easy wins. Either via the Add Field interface where the current "Text" option could be duplicated for Email, Number, Color (to cover some basics). Or as you suggest, a new drop-down added the the current "Text" Form Information interface where an Input Type could be changed from the default "text" to these others that would be suitable for single-line text input.

I hope someone dives in and has a crack at implementing :)
